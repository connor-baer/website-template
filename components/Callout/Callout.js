import { css } from '@emotion/core';
import styled from '@emotion/styled';

const baseStyles = ({ theme }) => css`
  text-align: center;
  margin: 0.5rem 0;
  padding: 2rem;
  border: 1px dashed ${theme.colors.p500};
  border-radius: ${theme.borderRadius.giga};
  font-size: 1.5rem;
  font-style: italic;
  line-height: 2.25rem;

  ${theme.mq.kilo} {
    grid-column: 1 / 13;
    padding: 2rem 4rem;
  }

  ${theme.mq.mega} {
    grid-column: 2 / 12;
    padding: 2rem 6rem;
  }
`;

const Callout = styled('div')(baseStyles);

export default Callout;
