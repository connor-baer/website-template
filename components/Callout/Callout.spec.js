import React from 'react';

import Callout from './Callout';

describe('Callout', () => {
  function renderCallout(fn, props = {}) {
    return fn(<Callout {...props} />);
  }

  it('should render with default styles', () => {
    const children = 'Bacon ipsum dolor amet swine shank biltong ham bacon.';
    const { container } = renderCallout(render, { children });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const children = 'Bacon ipsum dolor amet swine shank biltong ham bacon.';
    const html = renderCallout(renderToHtml, { children });
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
