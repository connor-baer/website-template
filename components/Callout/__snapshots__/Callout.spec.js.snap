// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Callout should render with default styles 1`] = `
.circuit-0 {
  text-align: center;
  margin: 0.5rem 0;
  padding: 2rem;
  border: 1px dashed #f06595;
  border-radius: 6px;
  font-size: 1.5rem;
  font-style: italic;
  line-height: 2.25rem;
}

@media (min-width:600px) {
  .circuit-0 {
    grid-column: 1 / 13;
    padding: 2rem 4rem;
  }
}

@media (min-width:900px) {
  .circuit-0 {
    grid-column: 2 / 12;
    padding: 2rem 6rem;
  }
}

<div
  class="circuit-0 circuit-1"
>
  Bacon ipsum dolor amet swine shank biltong ham bacon.
</div>
`;
